ompi/mca/sharedfp/sm/sharedfp_sm_file_open.c

```diff
@@ -231,6 +231,10 @@ int mca_sharedfp_sm_file_open (struct ompi_communicator_t *comm,
 
     comm->c_coll->coll_barrier (comm, comm->c_coll->coll_barrier_module );
 
+#if defined(HAVE_SEM_OPEN)
+    sem_unlink ( sm_data->sem_name);
```

Comment:
should this only be invoked on (per node ?) rank zero ? that is likely harmless, but this should be invoked only once per node.